tests/test_harmony.py

```diff
@@ -55,20 +55,16 @@ def run(random_state):
 
 
 def test_cluster_fn():
+    '''Test a custom cluster function.'''
     meta_data = pd.read_csv("data/pbmc_3500_meta.tsv.gz", sep="\t")
     data_mat = pd.read_csv("data/pbmc_3500_pcs.tsv.gz", sep="\t")
 
     def cluster_fn(data, K):
-        centroid, label = kmeans2(data, K, minit='++', seed=0)
+        centroid, label = kmeans2(data, K, minit='++')
         return centroid
 
-    def run(cluster_fn):
-        ho = hm.run_harmony(data_mat,
-                            meta_data, ['donor'],
-                            max_iter_harmony=2,
-                            max_iter_kmeans=2,
-                            cluster_fn=cluster_fn)
-        return ho.Z_corr
-
-    # Assert same results when random_state is set.
-    np.testing.assert_equal(run(cluster_fn), run(cluster_fn))
```

Comment:
I liked this line! This helps me to remember why we introduced this feature. The intention was to allow users to choose a perfectly deterministic kmeans if they do not like the default sklearn kmeans that is not perfectly deterministic.

Could we include this line? Or maybe include a special case for python 3.6 to get past the error about `seed=0`?